src/class/macros.rs

```diff
@@ -231,23 +220,20 @@ macro_rules! py_ternary_num_func {
         {
             use $crate::ObjectProtocol;
 
-            let py = $crate::Python::assume_gil_acquired();
-            let _pool = $crate::GILPool::new(py);
-            let arg1 = py.from_borrowed_ptr::<$crate::types::PyAny>(arg1);
-            let arg2 = py.from_borrowed_ptr::<$crate::types::PyAny>(arg2);
-            let arg3 = py.from_borrowed_ptr::<$crate::types::PyAny>(arg3);
-
-            let result = match arg1.extract() {
-                Ok(arg1) => match arg2.extract() {
-                    Ok(arg2) => match arg3.extract() {
-                        Ok(arg3) => $class::$f(arg1, arg2, arg3).into(),
-                        Err(e) => Err(e.into()),
-                    },
-                    Err(e) => Err(e.into()),
-                },
-                Err(e) => Err(e.into()),
-            };
-            $crate::callback::cb_convert($conv, py, result)
+            let gil = $crate::assume_gil_acquired();
+            let py = gil.py();
+            $crate::catch_unwind!(py, {
+                let _pool = $crate::GILPool::new(py);
+                let arg1 = py.from_borrowed_ptr::<$crate::types::PyAny>(arg1);
```

Comment:
How about `let arg1 = py.from_borrowed_ptr::<$crate::types::PyAny>(arg1).extract()?;`

Reply:
Good idea, let me try that...